AtDigitization/AtSimpleSimulation.h

```diff
@@ -64,7 +65,16 @@ class AtSimpleSimulation {
    void SetDistanceStep(double step) { fDistStep = step; } //<In mm
 
    void NewEvent();
-   void SimulateParticle(int Z, int A, const XYZPoint &iniPos, const PxPyPzEVector &iniMom);
+
+   /**
+    * Simulates a particle over a given distance and returns the position and momentum of the particle at the stoping
+    * point. By default the particle will stop when it reaches the end of the TPC. A user defined function can test the
+    * position and momentum of the particle for each time step and stop it when a given condition is met (such as a
+    * depth in the TPC or an energy to stop at).
+    */
+   std::pair<XYZPoint, PxPyPzEVector> SimulateParticle(
+      int Z, int A, const XYZPoint &iniPos, const PxPyPzEVector &iniMom,
+      std::function<bool(XYZPoint, PxPyPzEVector)> func = [](XYZPoint pos, PxPyPzEVector mom) { return true; });
```

Comment:
This function should take const references, but fine as is with a copy (it's only 7 numbers).

Reply:
The doxygen comment could also just reference the second `SimulateParticle` function, but say the difference (this requires we have added the model for this A/Z

Reply:
I'm not sure I agree with the doxygen comment since one function is public and the other is protected.

Reply:
My concern is someone changes some logic somewhere and doesn't update both dioxygen comments. I'd be fine with the second (protected) referencing the public too. I just don't like the duplication of information for long-term maintenance.

Reply:
I see what you mean. I think I've changed it to remove that potential future error.